assets/src/edit-story/app/media/utils/fetchRemoteFile.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { v4 as uuidv4 } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import getUrlExtension from './getUrlExtension';
+
+/**
+ * Helper to get remote file using fetch.
+ *
+ * @param {string} url URL of file.
+ * @param {string} mimeType Mime type of file.
+ * @return {Promise<File>} File object.
+ */
+async function fetchRemoteFile(url, mimeType) {
+  const name = uuidv4() + '.' + getUrlExtension(url);
```

Comment:
Why do we need a random file name here? Does this mean the video we're uploading will have a random file name? Doesn't seem ideal.

If the current video's name is `my-video.mp4`, I think we should be uploading something like `my-video-optimized.mp4` to WordPress instead.

Reply:
Fixed